fast64_internal/sm64/settings/properties.py

```diff
@@ -0,0 +1,174 @@
+import os
+import bpy
+from bpy.types import PropertyGroup, UILayout, Scene, Context
+from bpy.props import BoolProperty, StringProperty, EnumProperty, IntProperty, FloatProperty, PointerProperty
+from bpy.path import abspath
+from bpy.utils import register_class, unregister_class
+
+from ...render_settings import on_update_render_settings
+from ..sm64_constants import (
+    level_enums,
+    enum_refresh_versions,
+    defaultExtendSegment4,
+    enum_export_type,
+    enum_compression_formats,
+    enum_sm64_goal_type,
+)
+from ..sm64_utility import export_rom_checks, import_rom_checks
+from ...utility import (
+    directory_path_checks,
+    directory_ui_warnings,
+    multilineLabel,
+    prop_split,
+)
+
+
+def decomp_path_update(self, context: Context):
+    directory_path_checks(
+        abspath(self.decomp_path), "Empty decomp path.", "Decomp path does not exist.", "Decomp path is not a folder."
```

Comment:
raising an exception during a property update only displays it in the console afaik, it won't show to the user

so this just fills the console with long irrelevant traces

wrap in `try: except: return` ?

Reply:
good point